src/System.CommandLine.Benchmarks/CommandLine/Perf_Parser_CustomScenarios.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using BenchmarkDotNet.Attributes;
+
+namespace System.CommandLine.Benchmarks.CommandLine
+{
+    /// <summary>
+    /// Measures the performance of <see cref="System.CommandLine.Parser"/> for custom scenarios.
+    /// </summary>
+    [BenchmarkCategory(Categories.CommandLine)]
+    public class Perf_Parser_CustomScenarios
+    {
+        private string _testSymbolsAsString;
+        private Parser _testParser;
+
+        [GlobalSetup(Target = nameof(Scenario1_Parse))]
+        public void SetupScenario1()
+        {
+            var rootCommand = new Command("root_command");
+            var nestedCommand = new Command("nested_command");
+            nestedCommand.AddOption(new Option("-opt1",
+                                               "description...",
+                                               new Argument<int>(defaultValue: 123)
+                                               {
+                                                   Arity = ArgumentArity.ExactlyOne
+                                               }));
+            rootCommand.AddCommand(nestedCommand);
+
+            _testParser = new Parser(rootCommand);
+            _testSymbolsAsString = "root_command nested_command -opt1 321";
+        }
+
+        [Benchmark]
+        public ParseResult Scenario1_Parse() => _testParser.Parse(_testSymbolsAsString);
```

Comment:
Can this have a better name? Perf_Parser_CustomScenarios.OneOptWithNestedCommand or something. In the performance reporting system we use the method names as the display names in reporting. Assuming these tests wind up there some day it'd be nice if they didn't change names.

Reply:
Ok. Renamed it to `Perf_Parser_CustomScenarios.OneOptWithNestedCommand_Parse`.